app/client/components/productSwitch/CancellationSwitchConfirmed.tsx

```diff
@@ -0,0 +1,274 @@
+import { css } from '@emotion/react';
+import {
+	space,
+	neutral,
+	headline,
+	textSans,
+	brand,
+	brandAlt,
+} from '@guardian/source-foundations';
+import {
+	Button,
+	Stack,
+	SvgArrowRightStraight,
+} from '@guardian/source-react-components';
+import { useNavigate, Link, useLocation, Navigate } from 'react-router-dom';
+import { minWidth } from '../../styles/breakpoints';
+import GridPicture from '../images/GridPicture';
+import GridImage from '../images/GridImage';
+import { useContext } from 'react';
+import {
+	ProductSwitchContext,
+	ProductSwitchContextInterface,
+	ProductSwitchResponse,
+} from './productSwitchApi';
+import { getGeoLocation } from '../../geolocation';
+import {
+	getIosAppUrl,
+	productStartDate,
+	productFirstPaymentAmount,
+} from './productSwitchHelpers';
+import { CancellationRouterState } from '../cancel/CancellationContainer';
+import { headingCss, listCss, standfirstCss } from './productSwitchStyles';
+
+const CancellationSwitchConfirmed = () => {
+	const navigate = useNavigate();
+
+	const productSwitchContext = useContext(
+		ProductSwitchContext,
+	) as ProductSwitchContextInterface;
+	const location = useLocation();
+	const routerState = location.state as CancellationRouterState;
+	const productSwitchConfirmationInfo =
+		routerState?.productSwitchConfirmationInfo as ProductSwitchResponse;
+
+	if (!productSwitchConfirmationInfo) {
+		return <Navigate to="/" />;
```

Comment:
Is this the correct place to bung people off to? Depends on the reason why someone has tried to get to this url directly